From f8d217615e9ca9efcd836c299f9b86ea3e500c3c Mon Sep 17 00:00:00 2001
From: Michael Bestas <mikeioannina@cyanogenmod.org>
Date: Sun, 4 Dec 2016 21:05:56 +0200
Subject: [PATCH 17/25] legacy: Readd support for old perfd socket

* This is still used by older 8974 & 8916 blobs and possibly others
* This essentially reverts 3173a8c8b8a93e55403f65e60b31d937ea8a18b4

Change-Id: I5c51052b3622e63a14f74600cdbb76d515001841
---
 legacy-common/file_contexts | 1 +
 legacy-common/perfd.te      | 3 ++-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/legacy-common/file_contexts b/legacy-common/file_contexts
index 1bf6434..88b5882 100644
--- a/legacy-common/file_contexts
+++ b/legacy-common/file_contexts
@@ -2,6 +2,7 @@
 /(vendor|system/vendor)/bin/perfd     u:object_r:perfd_exec:s0
 /data/system/perfd(/.*)?              u:object_r:mpctl_data_file:s0
 /dev/socket/perfd                     u:object_r:mpctl_socket:s0
+/data/misc/perfd(/.*)?                u:object_r:mpctl_socket:s0
 
 # Location
 /data/misc/location(/.*)?                                           u:object_r:location_data_file:s0
diff --git a/legacy-common/perfd.te b/legacy-common/perfd.te
index 03fe599..2bd680a 100644
--- a/legacy-common/perfd.te
+++ b/legacy-common/perfd.te
@@ -30,7 +30,8 @@ allow perfd sysfs_lib:file w_file_perms;
 r_dir_file(perfd, sysfs_kgsl)
 
 # mpctl socket
-allow perfd mpctl_socket:sock_file rw_file_perms;
+allow perfd mpctl_socket:dir rw_dir_perms;
+allow perfd mpctl_socket:sock_file create_file_perms;
 
 # default_values file
 allow perfd mpctl_data_file:dir rw_dir_perms;
-- 
2.7.4

